### PR TITLE
fix: should never get cutomer price on purchase document

### PR DESCRIPTION
**Pull request by FHenry**
In some case in Purchase Order the field Customer is set with the customer coming from 
`Material Request `
or 
`erpnext/erpnext/stock/doctype/material_request/material_request.make_purchase_order_based_on_supplier`

and in this case the get_item_detail and _in fine_ get_item_price 

```
	if args.get("customer"):
			conditions += " and customer=%(customer)s"
        elif args.get("supplier"):
	        conditions += " and supplier=%(supplier)s"
```

Then customer win... on Purchase document this is a non sense

### CODE REVIEW

File: erpnext/stock/get_item_details.py [hunk 1]
```diff
@@ -88,8 +88,20 @@ def get_item_details(args, doc=None, for_validate=False, overwrite_warehouse=Tru
...
 	update_party_blanket_order(args, out)
 
+	current_customer = args.customer
+	# Never try to find a supplier price if customer is set
+	# these Doctype
+	if (
+		args.get("doctype") == "Purchase Order"
+		or args.get("doctype") == "Purchase Receipt"
+		or args.get("doctype") == "Purchase Invoice"
+	):
```
Comment (s-aga-r) [inline erpnext/stock/get_item_details.py]: Use `in` keyword to check the doctype:

`if (args.get("doctype") in ["Purchase Order", "Purchase Receipt", "Purchase Invoice"]):`

**Review by s-aga-r — CHANGES_REQUESTED** [review]

**Comment by deepeshgarg007** [comment]
@FHenry Can you please add a test case for this scenario as well so that such an issue is not repeated again

**Comment by FHenry** [comment]
> @FHenry Can you please add a test case for this scenario as well so that such an issue is not repeated again

@deepeshgarg007 Ok, I'm on it

**Comment by FHenry** [comment]
> @FHenry Can you please add a test case for this scenario as well so that such an issue is not repeated again

@deepeshgarg007 ok test is done

**Comment by FHenry** [comment]
@Mergifyio backport version-13-hotfix

**Comment by FHenry** [comment]
@Mergifyio backport version-14-hotfix

**Comment by FHenry** [comment]
@deepeshgarg007, @s-aga-r 
 Is it possible to add backport label it to  version-13-hotfix and version-14-hotfix or the process is make other PRs ?

EDIT: Sorry I didn't see the you've already put version-14-hotfix, the question remain for version-13-hotfix

**Pull request merged**